olive/engine/engine.py

```diff
@@ -420,68 +446,86 @@ def run_no_search(
 
-        # run all the passes in the step
-        (
-            _,
-            signal,
-            model_ids,
-        ) = self._run_passes(next_step["passes"], model, model_id, data_root, accelerator_spec)
-        # names of the output models of the passes
-        pass_output_names = [self.passes[pass_name]["output_name"] for pass_name, _ in next_step["passes"]]
-        pass_output_names = [f"{name}_{accelerator_spec}" if name else None for name in pass_output_names]
-
-        final_output_name = pass_output_names[-1]
-        if output_name:
-            # override the output name of the last pass
-            logger.debug("Engine output_name is provided. Will ignore output_name for final pass")
-            final_output_name = f"{output_name}_{accelerator_spec}"
-        elif not final_output_name:
-            # use the default output name
-            final_output_name = str(accelerator_spec)
-        pass_output_names[-1] = final_output_name
-
-        output_model_json = None
-        for pass_output_name, pass_output_model_id in zip(pass_output_names, model_ids):
-            if not pass_output_name:
-                continue
-            output_model_json = cache_utils.save_model(
-                model_number=pass_output_model_id,
-                output_dir=output_dir,
-                output_name=f"{pass_output_name}_model",
-                overwrite=True,
-                cache_dir=self._config.cache_dir,
-            )
+            # get the model id of the first input model
+            model_id = next_step["model_id"]
+            if model_id == input_model_id:
+                model = input_model
+            else:
+                model = self._load_model(model_id)
 
-        # save the evaluation results to output_dir
-        if signal is not None:
-            result_name = f"{final_output_name}_metrics"
-            results_path = output_dir / f"{result_name}.json"
-            with open(results_path, "w") as f:
-                json.dump(signal.to_json(), f, indent=4)
+            logger.debug(f"Step no search with search point {next_step['search_point']} ...")
 
-        if output_model_json:
-            output = {"model": output_model_json}
-            if signal is not None:
-                output["metrics"] = signal
-        else:
-            output = None
+            # run all the passes in the step
+            (
+                _,
+                signal,
+                model_ids,
+            ) = self._run_passes(next_step["passes"], model, model_id, data_root, accelerator_spec)
+            # names of the output models of the passes
+            pass_output_names = [self.passes[pass_name]["output_name"] for pass_name, _ in next_step["passes"]]
+            pass_output_names = [f"{name}_{accelerator_spec}" if name else None for name in pass_output_names]
+
+            pass_flow = self.pass_flows[iter_num - 1]
+            result_prefix = "-".join(pass_flow)
```

Comment:
maybe we can use the pass_flow index (something like `flow-0`) instead of joining the whole flow? The name might be too long. 

Also, the file name in the whisper example https://github.com/microsoft/Olive/blob/main/examples/whisper/test_transcription.py#L70

Reply:
It may be not intuitive for user to find the corresponding passes if we use index. I switch it to a directory like:
output_dir / **pass_flow_joint_str** / filename.

Is that okay for you?

Reply:
This looks good to me. Thanks!